server/src/main/java/org/candlepin/controller/Entitler.java

```diff
@@ -186,6 +211,22 @@ public void adjustEntitlementQuantity(Consumer consumer, Entitlement ent,
                 data.setConsumer(consumer);
             }
         }
+        if (consumer.isCdk()) {
+            if (config.getBoolean(ConfigProperties.STANDALONE) ||
+                    poolCurator.listByOwner(consumer.getOwner(), new Date()).size() == 0) {
```

Comment:
Here we look up all pools just to call getSize(). It would probably be more efficient to add a method to the curator to ask if it has an active subscription. Then we avoid pulling all the pool data back from the database.